solutions/standard/variables.tf

```diff
@@ -0,0 +1,114 @@
+  description = "The region in which to provision KMS resources. If using existing KMS, set this to the region in which it is provisioned in."
+  default     = "us-south"
+}
+
+########################################################################################################################
+# Secrets Manager
+########################################################################################################################
+
+variable "secrets_manager_instance_name" {
+  type        = string
+  description = "The name to give the Secrets Manager instance that will be provisioned by this solution."
+  default     = "base-security-services-sm"
+}
+
+variable "service_plan" {
+  type        = string
+  description = "The service/pricing plan to use when provisioning a new Secrets Manager instance. Allowed values: 'standard' and 'trial'. Only used if `provision_sm_instance` is set to true."
+  default     = "standard"
+  validation {
+    condition     = contains(["standard", "trial"], var.service_plan)
+    error_message = "Allowed values for sm_service_plan are \"standard\" and \"trial\"."
+  }
+}
+
+variable "service_endpoints" {
+  # public-and-private until IBM Console connects to SM via private endpoints
+  type        = string
+  description = "The service endpoints to enable for all services deployed by this solution. Allowed values are `private` or `public-and-private`. If selecting `public-and-private`, communication to the instances will all be done over the public endpoints. Ensure to enable virtual routing and forwarding (VRF) in your account if using `private`, and that the terraform runtime has access to the the IBM Cloud private network."
+  default     = "public-and-private"
+  validation {
+    condition     = contains(["private", "public-and-private"], var.service_endpoints)
+    error_message = "The specified service_endpoints is not a valid selection. Allowed values are `private` or `public-and-private`."
+  }
+}
+
+variable "secret_manager_tags" {
+  type        = list(any)
+  description = "The list of resource tags that you want to associate with your Secrets Manager instance."
+  default     = []
+}
+
+########################################################################################################################
+# Key Protect
+########################################################################################################################
+
+variable "enable_kms_encryption" {
+  type        = bool
+  description = "Set this to true to control the encryption keys used to encrypt the data that you store in Secrets Manager. If set to false, the data that you store is encrypted at rest by using envelope encryption. For more details, see https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-mng-data&interface=ui#about-encryption."
+  default     = false
+}
+
+variable "existing_kms_instance_guid" {
+  type        = string
+  description = "The CRN of an existing KMS instance to use."
```

Comment:
Only needed if creating kms auth policy and if not providing an existing kmd key crn